src/fs/filesystem.ts

```diff
@@ -164,6 +164,8 @@ export class FileSystem {
    * Loads an existing file system from a CID.
    */
   static async fromCID(cid: CID, opts: FileSystemOptions = {}): Promise<FileSystem | null> {
+    if (!(cid instanceof CID)) throw new Error("Expected a CID object")
```

Comment:
I'm sorry for throwing you into JS/TS land which is horrible :P

`instanceof` is dangerous in JS, because if for whatever reason you'll end up with a duplicated copy of the multiformats library in your bundle, their `CID` classes are not `instanceof`-compatible with each other.

Here's an issue comment that shows the desired pattern to use instead: https://github.com/multiformats/js-cid/issues/111#issuecomment-645003427

I know, this is an annoyingly small thing and you want to get working on something bigger, it's just for you to get familiar with different parts of our toolchain + code. (Almost done :tada:)

Reply:
No. It's fine. You have been really helpful. 😀